extensions/geode-modules/src/main/java/org/apache/geode/modules/session/catalina/DeltaSession.java

```diff
@@ -242,12 +252,15 @@ public void setAttribute(String name, Object value, boolean notify) {
 
     checkBackingCacheAvailable();
 
-    synchronized (this.changeLock) {
+    synchronized (changeLock) {
       // Serialize the value
       byte[] serializedValue = serialize(value);
 
       // Store the attribute locally
-      if (this.preferDeserializedForm) {
+      if (preferDeserializedForm) {
+        if (notify) {
```

Comment:
This seems unnecessary, as the next line code (super.setAttribute(name, value, true);) would override and set the attribute value again.